### PR TITLE
RR-431 - Update Induction API model to DTO mappers

### PR DESCRIPTION
**Pull request by mattwills-moj-digital**
This PR introduces the mappers to convert from the Update Induction API request objects to the equivalent Update DTO ones. It's essentially a copy & paste of the Create Induction mappers, except with the addition of the `id` field.

### CODE REVIEW

File: domain/induction/src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/domain/induction/FutureWorkInterests.kt [hunk 1]
```diff
@@ -27,7 +27,7 @@ data class FutureWorkInterests(
 data class WorkInterest(
   val workType: WorkInterestType,
   val workTypeOther: String?,
-  val role: String,
+  val role: String?,
```
Comment (mattwills-moj-digital) [inline domain/induction/src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/domain/induction/FutureWorkInterests.kt:30]: Unrelated change (as discussed with @nathanrussell-moj-digital)

**Review by nathanrussell-moj-digital — APPROVED** [review]
What's not to love? ❤️
